packages/scandipwa/src/component/MyAccountInformation/MyAccountInformation.container.js

```diff
@@ -145,47 +140,40 @@ export class MyAccountInformationContainer extends PureComponent {
         }
 
         updateCustomerLoadingStatus(true);
+        try {
+            if (showPasswordChangeField) {
+                await this.handlePasswordChange({ password, newPassword });
+            }
 
-        await this.handleInformationChange({ firstname, lastname, taxvat });
+            if (showEmailChangeField) {
+                await this.handleEmailChange({ email, password });
+            }
 
-        if (showPasswordChangeField) {
-            await this.handlePasswordChange({ password, newPassword });
-        }
+            await this.handleInformationChange({ firstname, lastname, taxvat });
 
-        if (showEmailChangeField) {
-            await this.handleEmailChange({ email, password });
+            this.handleSuccessChange();
+        } catch (e) {
+            this.handleLockAccount(e);
         }
-
-        this.afterSubmit();
     }
 
-    afterSubmit() {
-        const { showSuccessNotification, updateCustomerLoadingStatus, showErrorNotification } = this.props;
+    handleSuccessChange() {
         const {
-            isErrorShow, isLocked, showEmailChangeField, showPasswordChangeField
+            showSuccessNotification, updateCustomerLoadingStatus
+        } = this.props;
+        const {
+            showEmailChangeField, showPasswordChangeField
         } = this.state;
 
-        if (!isErrorShow) {
-            updateCustomerLoadingStatus(false);
-
-            if (showEmailChangeField || showPasswordChangeField) {
-                this.handleLogout({ isFromEmailChange: true });
-            } else {
-                history.push({ pathname: appendWithStoreCode(ACCOUNT_URL) });
-            }
+        updateCustomerLoadingStatus(false);
 
-            showSuccessNotification('You saved the account information.');
+        if (showEmailChangeField || showPasswordChangeField) {
+            this.handleLogout({ isFromEmailChange: true });
         } else {
-            this.setState({ isErrorShow: false });
-
-            if (isLocked) {
-                const message = 'The account sign-in was incorrect or your account is disabled temporarily.'
-                + 'Please wait and try again later.';
-
-                showErrorNotification(message);
-                this.handleLogout({ isFromLocked: true });
-            }
+            history.push({ pathname: appendWithStoreCode(ACCOUNT_URL) });
         }
+
+        showSuccessNotification('You saved the account information.');
```

Comment:
Requires translation: `__('You saved the account information.')`

Reply:
done